### PR TITLE
Feature/tt 460 celo erc20

### PR DESCRIPTION
**Pull request by vlnevyhosteny**

### CODE REVIEW

File: packages/blockchain/celo/src/lib/transactions/erc20/index.ts [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  testnet?: boolean,
+  contractAddress?: string,
+) => {
+  const celoProvider = celoUtils.getProvider(provider)
+
+  return {
+    celoProvider,
+    network: await celoProvider.ready,
+    feeCurrencyContractAddress: celoUtils.getFeeCurrency(args.feeCurrency, testnet),
+    contract: new new Web3(
```
Comment (alexloiko) [inline packages/blockchain/celo/src/lib/transactions/erc20/index.ts:29]: new new?

File: packages/blockchain/celo/src/lib/transactions/erc20/index.ts [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  const { txCount, gasPrice, from } = await celoUtils.obtainWalletInformation(
+    wallet,
+    feeCurrencyContractAddress,
+  )
+
+  const tx: CeloTransactionConfig = {
+    chainId: network.chainId,
+    feeCurrency: feeCurrencyContractAddress,
+    nonce: nonce || txCount,
+    gasLimit: fee?.gasLimit ? '0x' + new BigNumber(fee.gasLimit).toString(16) : undefined,
```
Comment (alexloiko) [inline packages/blockchain/celo/src/lib/transactions/erc20/index.ts:179]: lets create util toHex

File: packages/shared/abstract-sdk/src/lib/utils.ts [hunk 3]
```diff
@@ -17,6 +17,8 @@ export const amountUtils = {
     }
     return Number(satoshis)
   },
+  multiplyToHexString: (amount: string, decimals: number) =>
```
Comment (alexloiko) [inline packages/shared/abstract-sdk/src/lib/utils.ts:20]: I would suggest to rename it to ```amountToHexString(amount: string, decimals: number)```

File: packages/shared/abstract-sdk/src/lib/utils.ts [hunk 4]
```diff
@@ -38,3 +40,5 @@ export const placeArgsToString = (message: string, args?: SdkMessageArgs): strin
   }
   return message
 }
+
+export const toHexString = (value: BigNumber) => `0x${value.toString(16)}`
```
Comment (alexloiko) [inline packages/shared/abstract-sdk/src/lib/utils.ts:44]: maybe let's move it under amount utils?

File: packages/blockchain/celo/src/lib/transactions/smartContract/index.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    feeCurrencyContractAddress,
+  )
+
+  transaction.nonce = transaction.nonce || txCount
+  transaction.from = from
+  transaction.gasLimit =
+    transaction.gasLimit ??
+    (await wallet.estimateGas(transaction)).add(feeCurrency === Currency.CELO ? 0 : 100000).toHexString()
+  transaction.gasPrice = fee?.gasPrice
+    ? '0x' + new BigNumber(toWei(fee.gasPrice, 'gwei')).toString(16)
```
Comment (alexloiko) [inline packages/blockchain/celo/src/lib/transactions/smartContract/index.ts:66]: please check all the places to use new util

File: packages/shared/abstract-sdk/src/lib/utils.ts (hunk 4 above)
Comment (vlnevyhosteny) [reply packages/shared/abstract-sdk/src/lib/utils.ts:44]: I think its general not really connected to amount. Using it for example in `gasLimitToHexWithFallback`.

**Review by alexloiko — APPROVED** [review]

**Pull request merged**